magda-typescript-common/src/authorization-api/authMiddleware.ts

```diff
@@ -16,27 +16,30 @@ export const mustBeAdmin = (baseAuthUrl: string, jwtSecret: string) => (
     next: () => void
 ) => {
     const rejectNoAuth = () => res.status(401).send("Not authorized.");
-    const apiClient = new ApiClient(baseAuthUrl);
 
     getUserIdHandling(req, res, jwtSecret, (userId: string) => {
-        apiClient.getUser(userId).then(maybeUser => {
-            maybeUser.caseOf({
-                just: user => {
-                    (req as any).user = user;
-                    if (user.isAdmin) {
+        const apiClient = new ApiClient(baseAuthUrl, jwtSecret, userId);
+
+        apiClient
```

Comment:
I'd really prefer if we kept the explicit `isAdmin` check in here - I see how this works now, but having permission to access `/private/users/{id}` isn't necessarily the same as being an admin. If we changed it at some point in the future so that some users who aren't admins can still access `/private/users/{id}`, or if someone changed `getUser()` to use the `/public` API, then non-admin users would automatically get admin access to every other service that uses this middleware.

Reply:
I think you are right. When I was working on it, I also thought it might be necessary to consolidate / re-structure the authorization related function. The session validation & permission checking should be move into a bigger version of ApiClients.ts (the API might needs to be extends a little bit too).
Here I just want to keep minimal impact & changes to existing code, so that we can have this issue closed up and move further from there. 
How about I create another issue regarding this and work on this after this issue?

Reply:
Yeah that sounds like a good idea! But can you leave the `user.isAdmin` check in here just in case we don't get around to doing the bigger story for a while? :)

Reply:
Have put the `isAdmin` checking back --- I think you are right --- It will be safer